tests/unit/test_context.py

```
@@ -934,3 +938,452 @@ def test_get_env_context_integration():
+    """Test graceful handling of permission errors."""
+    monkeypatch.chdir(tmp_path)
+
+    # Create a directory we'll make unreadable
+    subdir = tmp_path / "restricted"
+    subdir.mkdir()
+
+    # Mock scandir to raise PermissionError
+    original_scandir = os.scandir
+    def mock_scandir(path):
+        if str(path).endswith("restricted"):
+            raise PermissionError("Access denied")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)